test/TorchSharpTest/TestTorchTensor.cs

```diff
@@ -4459,6 +4459,27 @@ public void RoundTest()
             Assert.True(res.allclose(torch.tensor(expected)));
         }
 
+        [Fact]
+        public void RoundTestWithDecimals()
+        {
+            const long n = 7L;
+            var i = eye(n); // identity matrix
+            var a = rand(new[] { n, n });
+            var b = linalg.inv(a);
+
+            // check non-inline version
+            var r0 = round(matmul(a, b), 2L);
+            var r1 = round(matmul(b, a), 3L);
+            Assert.True(torch.Equals(i, r0), "round() failed");
```

Comment:
Any chance of rounding causing false assertions here? There's an allclose() comparator that I've been using in some unit tests when comparing tensor arithmetic to some expected outcome.